### PR TITLE
Rake

### PR DESCRIPTION
**Pull request by l8556**
Add rake

Commands for running:

rake download[all] - To download all files

rake download[file] - To download files. Reading the array of files comes from "./data/files_to_download.list"

rake download[ext,your extension] - To download files by extension. You must specify the extension, the second parameter.

rake download[arrext] - To download files by extensions from array.
The array is in the file "./data/static_data.rb". Change "EXTENSION_ARRAY"

rake download[arrfile] - To download files by file names from array. The array is in the file "./data/static_data.rb".
Change "FILE_NAMES_ARRAY".')

### CODE REVIEW

File: lib/main.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'onlyoffice_s3_wrapper'
+require_relative '../data/static_data'
+
+
+# Methods for download files
+class Downloader
+  def initialize
+    FileUtils.makedirs('./tmp')
```
Comment (ShockwaveNN) [inline lib/main.rb:10]: I think those lines can be switched and you can call `FileUtils.makedirs(@tmp_dir)` - so if we change this location in future - we need to change only one line

File: lib/main.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    @tmp_dir = './tmp'
+  end
+
+  def s3
+    @s3 ||= OnlyofficeS3Wrapper::AmazonS3Wrapper.new(bucket_name: 'conversion-testing-files', region: 'us-east-1')
+  end
+
+  def download(lib)
+    lib.each do |el|
```
Comment (ShockwaveNN) [inline lib/main.rb:19]: `lib` and `el` bad names for variables, 

`lib` is misleading
`el` is too short and can be replaced to another name

File: lib/main.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      p("Starting downloading file: #{el}")
+      s3.download_file_by_name(el, @tmp_dir)
+    end
+  end
+
+  def download_all
+    filename = s3.get_files_by_prefix
```
Comment (ShockwaveNN) [inline lib/main.rb:26]: `filename` is bad name

File: lib/main.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    download(filename_from_txt)
+  end
+
+  def download_by_extension(extension)
+    filename = s3.get_files_by_prefix("#{extension}/")
+    download(filename)
+  end
+
+  def download_by_array_extension
+    StaticData::EXTENSION_ARRAY.each do |el|
```
Comment (ShockwaveNN) [inline lib/main.rb:47]: `el` bad name

File: lib/validation/validator.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      Downloader.new.download_from_file
+    when 'ext'
+      puts('Please, enter extension. Example: rake download[ext,<extension>]') if extension == ''
+      Downloader.new.download_by_extension(extension)
+    when 'arrext'
+      Downloader.new.download_by_array_extension
+    when 'arrfile'
+      Downloader.new.download_by_array_filenames
+    else
+      puts('Input Error
```
Comment (ShockwaveNN) [inline lib/validation/validator.rb:19]: Better to split lines something like this:

```ruby
message =  'Input Error' \
                    'Please,enter the correct parameters' \
                    'Example: rake download[parameter,extension]`
puts(message)                 
```
It's much simplier to read

File: lib/validation/validator.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# use Validator.new.validator_for_download
+class Validator
+  def validator_for_download(download_flag, extension)
+    case download_flag
+    when 'all'
```
Comment (ShockwaveNN) [inline lib/validation/validator.rb:7]: I think those values 'all', 'file' etc should be symbols - `:all`, :file` etc...

Read about symbols in ruby - it's rather uniq and intersting difference between string and symbol and as far as I'm aware - there is no such think in python as symbols

**Review by ShockwaveNN — CHANGES_REQUESTED** [review]

**Review by ShockwaveNN — APPROVED** [review]

**Review by askonev — APPROVED** [review]

**Pull request merged**